config/compression.go

```diff
@@ -0,0 +1,81 @@
+package config
+
+import "fmt"
+
+type Compression struct {
+	Request  ReqCompressionConfig  `mapstructure:"request"`
+	Response RespCompressionConfig `mapstructure:"response"`
+}
+
+// CompressionConfig defines if compression is enabled and what type of compression to use
+type ReqCompressionConfig struct {
+	Enabled bool     `mapstructure:"enabled"`
+	Kind    []string `mapstructure:"kind,flow"`
+	kindMap map[CompressionKind]bool
```

Comment:
With only one currently supported compression kind, so you think we need the map?

Reply:
I agree that for just one supported compression kind we don't need the map but this sets us up for the future if we do end up supporting more compression types without adding too much overhead.